doc/usage.rst

```diff
@@ -109,6 +114,37 @@ On initialization, the PennyLane-Forest devices accept additional keyword argume
     on a provided QMI, these environment variables are set automatically and will also
     not need to be passed in PennyLane.
 
+.. note::
+
+    If using the pyQuil built-in pyQVM, you will still need to have an accessible Forest Quil compiler
+    server. This must be launched with the command ``quilc -S -P``, where the ``-P`` flag restricts
+    the input/output of the compiler to protoquil.
+
+
+The ``forest.numpy`` device
+===========================
+
+The ``forest.numpy`` device provides an interface between PennyLane and the pyQuil `NumPy wavefunction simulator <http://docs.rigetti.com/en/stable/wavefunction_simulator.html>`_. Because the NumPy wavefunction simulator allows access and manipulation of the underlying quantum state vector, ``forest.numpy`` is able to support the full suite of PennyLane and Quil quantum operations and expectation values.
+
+In addition, it is generally faster than running equivalent simulations on the QVM, as the final state can be inspected and the expectation value calculated analytically, rather than by sampling measurements.
```

Comment:
The QVM allows computing an expectation value with the `.expectation` call. The NumPy simulator lets you, within Python, inspect the wavefunction without it being transmitted. Generally, the NumPy simulator will be faster for small numbers of qubits or very rapid evaluations of many programs on a small number of qubits, since there's no interprocess communication overhead.

See the [`expectation` method](https://github.com/rigetti/pyquil/blob/master/pyquil/api/_wavefunction_simulator.py#L77). This gets computed analytically.

I suggest making this paragraph more nuanced than it is now.

Reply:
Not within the scope of this PR, but at some point we might also want to have Pennylane-Forest talk to `operator_estimation`, particularly the [`measure_observables(..)` function](https://github.com/rigetti/pyquil/blob/c41a3bd446a6a55e06fe1bb252358f0cbd145c4f/pyquil/operator_estimation.py#L751).

Reply:
I agree - I'll create a separate issue and a new PR for this.

Reply:
Indeed, the `WavefunctionSimulator` and `QVMConnection` both calculate `expectation` by simply using dot products. They would sample if the program instruction contained some noise operations however (see https://github.com/rigetti/pyquil/blob/master/pyquil/api/_qvm.py#L247)

Reply:
Created an issue: https://github.com/rigetti/pennylane-forest/issues/9